src/UI/Internal/Filter/View.elm

```diff
@@ -0,0 +1,520 @@
+                , Element.padding 10
+                ]
+                rows
+            , bodyButtons renderConfig common.size buttons
+            ]
+    in
+    Element.column attrs
+        bodyRows
+        |> Element.clickElsewhereToLeave closeMsg []
+
+
+bodyHeader : RenderConfig -> FilterSize -> msg -> String -> Element msg
+bodyHeader renderConfig size closeMsg label =
+    let
+        { padding, fontSize, iconSize } =
+            headerProportions size
+
+        paddingWithBorders =
+            { left = padding.left + 2
+            , top = padding.top + 1
+            , bottom = padding.bottom + 1
+            , right = padding.right
+            }
+    in
+    Element.row
+        [ Element.paddingEach paddingWithBorders
+        , Element.width fill
+        , Border.color Colors.gray300
+        , Border.widthEach { zeroPadding | bottom = 1 }
+        , Font.color Colors.navyBlue700
+        , Font.size fontSize
+        , Font.semiBold
+        ]
+        [ Element.text label
+        , bodyCloseIcon renderConfig closeMsg iconSize
+        ]
+
+
+bodyCloseIcon : RenderConfig -> msg -> Int -> Element msg
+bodyCloseIcon renderConfig closeMsg iconSize =
+    RenderConfig.localeTerms renderConfig
+        |> (.filters >> .close)
+        |> Icon.close
+        |> Icon.withCustomSize iconSize
+        |> Icon.renderElement renderConfig
+        |> Element.el
+            (ARIA.toElementAttributes ARIA.roleButton
+                ++ [ Element.alignRight
+                   , Element.pointer
+                   , Events.onClick closeMsg
+                   , Element.onEnterPressed closeMsg
+                   , Element.tabIndex 0
+                   ]
+            )
+
+
+bodySorting : RenderConfig -> FilterSize -> Maybe (BodySorting msg) -> Element msg
+bodySorting renderConfig size sorting =
+    case sorting of
+        Just sortingData ->
+            let
+                proportions =
+                    bodySortingProportions size
+
+                terms =
+                    RenderConfig.localeTerms renderConfig
+            in
+            Element.column [ Element.width fill ]
+                [ sortAs renderConfig
+                    SortAscending
+                    proportions
+                    sortingData
+                    terms.tables.sorting.ascending
+                    sortingData.ascendingSortMsg
+                , sortAs renderConfig
+                    SortDescending
+                    proportions
+                    sortingData
+                    terms.tables.sorting.descending
+                    sortingData.descendingSortMsg
+                ]
+
+        Nothing ->
+            Element.none
+
+
+sortAs :
+    RenderConfig
+    -> SortingDirection
+    -> { fontSize : Int, iconSize : Int, padding : RectangleSides }
+    -> BodySorting msg
+    -> String
+    -> msg
+    -> Element msg
+sortAs renderConfig direction { fontSize, iconSize } sortingData label msg =
```

Comment:
I think this could be a single record

Reply:
Sorry, I disagree:
1 - It's not confusing as every entry has a different type;
2 - Wrongly inserting one argument would break the compilation;
3 - A record would increase overhead;
4 - A record would break the capability for partially applying arguments;